pgmpy/sampling/Sampling.py

```diff
@@ -158,6 +160,7 @@ def rejection_sample(self, evidence=None, size=1, return_type="dataframe"):
 
             i += len(_sampled)
 
+        sampled = _map_to_state_name(self.model, sampled)
```

Comment:
I would prefer `_map_to_state_name` to be from `return_samples`. In my opinion it belongs there, since `_return_samples` is supposed to postprocess the samples. Also, with that we will not need to call this method everywhere.

Reply:
@ankurankan No. return_samples is also being used by hmc and nuts. So it will make code messy IMO.

Reply:
@khalibartan Yeah, but `return_samples` is supposed to be like a post processor, so any cases regarding that should be handled inside it. And if implementing that messes up the code too much, we should have a different solution to that, maybe implementing 2 separate post processors for different cases. But calling 2 post processing methods doesn't make much sense.

Reply:
@ankurankan I'm in favour of keeping these two separately. I don't see any problem in this and it keeps the code clean.

Reply:
@ankurankan What should be done here?  Both functions have different work.Two different post processors for the different case will lead to redundancy. Merging them into one can be done if we want to.

Reply:
@khalibartan I do think that we should merge these two.